mdbenchmark/plot.py

```diff
@@ -0,0 +1,246 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDBenchmark
+# Copyright (c) 2017 Max Linke & Michael Gecht and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# MDBenchmark is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# MDBenchmark is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MDBenchmark.  If not, see <http://www.gnu.org/licenses/>.
+import pandas as pd
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
+from matplotlib.figure import Figure
+import click
+
+from .cli import cli
+from .utils import calc_slope_intercept, lin_func, generate_output_name
+from . import console
+
+
+def plot_line(df, df_sel, label, ax=None):
+    if ax is None:
+        ax = plt.gca()
+
+    p = ax.plot(df_sel, 'ns/day', '.-', data=df, ms='10', label=label)
+    color = p[0].get_color()
+    slope, intercept = calc_slope_intercept(
+        (df[df_sel].iloc[0], df['ns/day'].iloc[0]), (df[df_sel].iloc[1],
+                                                     df['ns/day'].iloc[1]))
+    # avoid a label and use values instead of pd.Series
+    ax.plot(
+        df[df_sel],
+        lin_func(df[df_sel].values, slope, intercept),
+        ls='--',
+        color=color,
+        alpha=.5)
+    return ax
+
+
+def plot_over_group(df, plot_cores, ax=None):
+    # plot all lines
+    df_sel = 'ncores' if plot_cores else 'nodes'
+
+    gb = df.groupby(['gpu', 'module', 'host'])
+    groupby = ['gpu', 'module', 'host']
+    for key, df in gb:
+        label = ' '.join(['{}={}'.format(n, v) for n, v in zip(groupby, key)])
```

Comment:
We need to think about the label text. The current approach isn't great.

Reply:
agreed, but I wouldn't leave out any of the information if possible. Maybe it would be possible to move the labels into a box below the plot in form of a little table or something? Or maybe one could even print the dataframe below as reference?

Reply:
Have a look at the seaborn.FacetGrid code. It can place the legends outside the plotting area as standard.

Reply:
https://github.com/mwaskom/seaborn/blob/67126155553f8bc014fc178ae9a812f604efaa25/seaborn/axisgrid.py#L83

Reply:
Yes, we should probably use something like that. I recently had the same problem with one of my plots and stumbled upon this excellent answer on StackOverflow reviewing all possibilities with `matplotlib`: https://stackoverflow.com/a/43439132/4791226